provider/openstack/provider.go

```diff
@@ -1619,7 +1623,9 @@ func (e *Environ) configureRootDisk(_ context.ProviderCallContext, args environs
 	}
 	switch rootDiskSource {
 	case rootDiskSourceLocal:
-		runOpts.ImageId = spec.Image.Id
+		if !args.Constraints.HasImageID() {
```

Comment:
I don't know why the provider has this different cases, but I had to add the check on the constraint, because otherwise the image-id is never used when we are on a local disk (this is the case of my local tests for example) and a user passes the image-id constraint.